### PR TITLE
tiproxy: add vip configurations

### PR DESCRIPTION
**Pull request by djshow832**
### What is changed, added or deleted? (Required)

1. Add virtual IP-related doc.
2. Remove Prometheus-related limitations in load balance. If Prometheus is unavailable, TiProxy will query TiDB directly.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v8.3 (TiDB 8.3 versions)
- [ ] v8.2 (TiDB 8.2 versions)
- [ ] v8.1 (TiDB 8.1 versions)
- [ ] v8.0 (TiDB 8.0 versions)
- [ ] v7.5 (TiDB 7.5 versions)
- [ ] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Review by joey-yez — APPROVED** [review]

File: tiproxy/tiproxy-configuration.md [hunk 1]
```diff
@@ -15,8 +15,9 @@ max-connections = 100
 [api]
 addr = "0.0.0.0:3080"
 
-[log]
-level = "info"
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:19]: Is there a specific reason for removing `log.level` from the config example?

File: tiproxy/tiproxy-configuration.md [hunk 2]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
 + 可选值：`resource`、`location`、`connection`
 + 指定负载均衡策略。各个可选值的含义请参阅 [TiProxy 负载均衡策略](/tiproxy/tiproxy-load-balance.md#负载均衡策略配置)。
 
+### `ha`
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:123]:
```suggestion
### ha
```

File: tiproxy/tiproxy-configuration.md [hunk 3]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
+
+TiProxy 的高可用配置。
+
+#### `virtual-ip`
+
++ 默认值：``
++ 支持热加载：否
++ 指定虚拟 IP 地址，使用 CIDR 格式，例如 `10.0.1.10/24`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，其他 TiProxy 会绑定该 IP，确保客户端使用该虚拟 IP 总能连接到可用的 TiProxy。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:131]:
```suggestion
+ 指定虚拟 IP 地址，使用 CIDR 格式表示，例如 `10.0.1.10/24`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，其他 TiProxy 会绑定该 IP，确保客户端始终能通过虚拟 IP 连接到可用的 TiProxy。
```

File: tiproxy/tiproxy-configuration.md [hunk 4]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
+
+> **注意：**
+>
+> - 虚拟 IP 仅支持 Linux 操作系统。
+> - 请确保运行 TiProxy 的 Linux 用户具有绑定 IP 地址的权限。
+> - 请确保虚拟 IP 和所有 TiProxy 的 IP 在同一个 CIDR 段内。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:137]:
```suggestion
> - 运行 TiProxy 的 Linux 用户必须具有绑定 IP 地址的权限。
> - 虚拟 IP 和所有 TiProxy 实例的 IP 必须处于同一个 CIDR 范围内。
```

File: tiproxy/tiproxy-configuration.md [hunk 5]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
+
+#### `interface`
+
++ 默认值：``
++ 支持热加载：否
++ 指定绑定虚拟 IP 的网络接口，例如 `eth0`。仅当 `virtual-ip` 和 `interface` 同时设置时，该实例才有可能绑定虚拟 IP。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-configuration.md:143]:
```suggestion
+ 指定绑定虚拟 IP 的网络接口，例如 `eth0`。只有同时设置 [`ha.virtual-ip`](#virtual-ip) 和 `ha.interface` 时，该 TiProxy 实例才能绑定虚拟 IP。
```

File: tiproxy/tiproxy-deployment-topology.md [hunk 6]
```diff
@@ -32,6 +32,12 @@ global:
   ssh_port: 22
   deploy_dir: "/tidb-deploy"
   data_dir: "/tidb-data"
+component_versions:
+    tiproxy: "v1.2.0"
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-deployment-topology.md:36]:
```suggestion
  tiproxy: "v1.2.0"
```

File: tiproxy/tiproxy-overview.md [hunk 7]
```diff
@@ -91,7 +91,7 @@ TiProxy 不适用于以下场景：
 
 3. 配置 TiProxy 实例。
 
-    为了保证 TiProxy 的高可用，建议部署至少 2 台 TiProxy 实例。可以通过硬件负载均衡使流量分发到各 TiProxy 实例上，或配置虚拟 IP 使流量路由到可用的 TiProxy 实例上。
+    为了保证 TiProxy 的高可用，建议部署至少 2 台 TiProxy 实例，并配置虚拟 IP 使流量路由到可用的 TiProxy 实例上。
```
Comment (Oreoxmt) [inline tiproxy/tiproxy-overview.md:94]:
```suggestion
    为了保证 TiProxy 的高可用，建议部署至少 2 台 TiProxy 实例，并配置虚拟 IP（[`ha.virtual-ip`](/tiproxy/tiproxy-configuration.md#virtual-ip) 和 [`ha.interface`](/tiproxy/tiproxy-configuration.md#interface)）使流量路由到可用的 TiProxy 实例上。
```

File: tiproxy/tiproxy-configuration.md (hunk 1 above)
Comment (djshow832) [reply tiproxy/tiproxy-configuration.md:19]: It's just an example. I don't want to make the example too long.
`log.level` is `info` by default, but typically users need to set `virtual-ip`.

**Review by Oreoxmt — APPROVED** [review]

File: tiproxy/tiproxy-configuration.md [hunk 8]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
 
+### ha
+
+TiProxy 的高可用配置。
+
+#### `virtual-ip`
+
++ 默认值：``
++ 支持热加载：否
++ 指定虚拟 IP 地址，使用 CIDR 格式表示，例如 `10.0.1.10/24`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，其他 TiProxy 会绑定该 IP，确保客户端始终能通过虚拟 IP 连接到可用的 TiProxy。
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:131]:
```suggestion
+ 指定虚拟 IP 地址，使用 CIDR 格式表示，例如 `10.0.1.10/24`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，另外一台 TiProxy 会自动绑定该 IP，确保客户端始终能通过虚拟 IP 连接到可用的 TiProxy。
```

Comment (qiancai) [reply tiproxy/tiproxy-configuration.md:131]: 不确定一台下线后，另外一台是否会自动绑定该 IP 呢，如果是，可以加上 “自动”

**Review by qiancai — APPROVED** [review]
Rest LGTM

File: tiproxy/tiproxy-configuration.md [hunk 9]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
 + 可选值：`resource`、`location`、`connection`
 + 指定负载均衡策略。各个可选值的含义请参阅 [TiProxy 负载均衡策略](/tiproxy/tiproxy-load-balance.md#负载均衡策略配置)。
 
+### ha
+
+TiProxy 的高可用配置。
+
+#### `virtual-ip`
+
++ 默认值：``
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:129]: 这里需要修改下吗
image
```suggestion
+ 默认值：""
```

File: tiproxy/tiproxy-configuration.md [hunk 10]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
+
+> **注意：**
+>
+> - 虚拟 IP 仅支持 Linux 操作系统。
+> - 运行 TiProxy 的 Linux 用户必须具有绑定 IP 地址的权限。
+> - 虚拟 IP 和所有 TiProxy 实例的 IP 必须处于同一个 CIDR 范围内。
+
+#### `interface`
+
++ 默认值：``
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:141]:
```suggestion
+ 默认值：""
```

File: tiproxy/tiproxy-configuration.md [hunk 11]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
 
+### ha
+
+TiProxy 的高可用配置。
+
+#### `virtual-ip`
+
++ 默认值：`""`
++ 支持热加载：否
++ 指定虚拟 IP 地址，使用 CIDR 格式表示，例如 `10.0.1.10/24`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，另外一台 TiProxy 会自动绑定该 IP，确保客户端始终能通过虚拟 IP 连接到可用的 TiProxy。
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:131]:
```suggestion
+ 指定虚拟 IP 地址，使用 CIDR 格式表示，例如 `"10.0.1.10/24"`。当集群中部署了多台 TiProxy 时，只有一台 TiProxy 会绑定虚拟 IP。当该 TiProxy 下线时，另外一台 TiProxy 会自动绑定该 IP，确保客户端始终能通过虚拟 IP 连接到可用的 TiProxy。
```

File: tiproxy/tiproxy-configuration.md [hunk 12]
```diff
@@ -119,6 +120,28 @@ TiProxy 负载均衡策略的配置。
...
+>
+> - 虚拟 IP 仅支持 Linux 操作系统。
+> - 运行 TiProxy 的 Linux 用户必须具有绑定 IP 地址的权限。
+> - 虚拟 IP 和所有 TiProxy 实例的 IP 必须处于同一个 CIDR 范围内。
+
+#### `interface`
+
++ 默认值：`""`
++ 支持热加载：否
++ 指定绑定虚拟 IP 的网络接口，例如 `eth0`。只有同时设置 [`ha.virtual-ip`](#virtual-ip) 和 `ha.interface` 时，该 TiProxy 实例才能绑定虚拟 IP。
```
Comment (qiancai) [inline tiproxy/tiproxy-configuration.md:143]:
```suggestion
+ 指定绑定虚拟 IP 的网络接口，例如 `"eth0"`。只有同时设置 [`ha.virtual-ip`](#virtual-ip) 和 `ha.interface` 时，该 TiProxy 实例才能绑定虚拟 IP。
```

**Pull request merged**